flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/config/KinesisStreamsSourceConfigConstants.java

```diff
@@ -57,4 +69,20 @@ public enum InitialPosition {
                     .longType()
                     .defaultValue(10000L)
                     .withDescription("The interval between each attempt to discover new shards.");
+
+    public static final ConfigOption<ReaderType> READER_TYPE =
+            ConfigOptions.key("type")
+                    .enumType(ReaderType.class)
+                    .defaultValue(ReaderType.POLLING)
+                    .withDescription("The type of reader used to read from the Kinesis stream.");
+
+    public static final ConfigOption<ConsumerLifecycle> EFO_CONSUMER_LIFECYCLE =
+            ConfigOptions.key("efo.consumer.lifecycle")
```

Comment:
Propose to use `kinesis.stream.efo.consumer.lifecycle` as the config key. As I will be updating the public example to use this.